### PR TITLE
Solve conflict between ACI and base methods in MapieTimeSeriesRegressor

### PR DESCRIPTION
**Pull request by BaptisteCalot**
# Description

This PR is to fix issue 447 where the method of MapieTimesSeriesRegressor is "aci" but when we use the fit method of MapieRegressor, the method become "base"

## Type of change

- Bug fix : In MapieRegressor, when cv is prefit or split and method different from "naive" or "plus", you don't force the method to become "base"

# How Has This Been Tested?

- [x] re-create example and make sure it works:
- [x] Build instance of MapieTimeSeriesRegressor with method "aci"
- [x] Fit instance (with fit method from MapieRegressor)
- [x] Update the instance
- [x] Finaly check the instance's method and see if it still "aci"

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: mapie/regression/regression.py [hunk 1]
```diff
@@ -425,7 +425,8 @@ def _check_fit_parameters(
         cv = check_cv(
             self.cv, test_size=self.test_size, random_state=self.random_state
         )
-        if self.cv in ["split", "prefit"] and self.method != "base":
+        if self.cv in ["split", "prefit"] and (self.method == "naive" or
+                                               self.method == "plus"):
```
Comment (thibaultcordier) [inline mapie/regression/regression.py:429]:
```suggestion
        if self.cv in ["split", "prefit"] and \
            self.method in ["naive", "plus", "minmax]:
```

**Review by thibaultcordier — CHANGES_REQUESTED** [review]
Thank you for your contribution @BaptisteCalot ! I suggested that you also take into account the ‘minmax’ method, which can cause problems such as `plus` and `naive`.

File: mapie/tests/test_time_series_regression.py [hunk 2]
```diff
@@ -518,3 +518,21 @@ def test_method_error_in_update(monkeypatch: Any, method: str) -> None:
     with pytest.raises(ValueError, match=r".*Invalid method.*"):
         mapie_ts_reg.fit(X_toy, y_toy)
         mapie_ts_reg.update(X_toy, y_toy)
+
+
+def test_aci_method_in_fit() -> None:
```
Comment (LacombeLouis) [inline mapie/tests/test_time_series_regression.py:523]: You should probably see if you can make the test more generic... for example, `"enbpi"` should also not have its method changed right?

**Comment by LacombeLouis** [comment]
Hey @BaptisteCalot,
Thank you for your contribution! Make sure to make the tests as generic as possible to check for possible other bugs.

**Comment by LacombeLouis** [comment]
Hey @BaptisteCalot, thank you for the generalized test! Make sure to tick all the boxes! Don't forget to:
- [ ] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files

File: mapie/tests/test_time_series_regression.py [hunk 3]
```diff
@@ -518,3 +518,25 @@ def test_method_error_in_update(monkeypatch: Any, method: str) -> None:
...
+    )
+    estimator = LinearRegression().fit(X_train, y_train)
+    mapie_ts_reg = MapieTimeSeriesRegressor(
+        estimator=estimator,
+        cv=cv, method=method
+    )
+    mapie_ts_reg.fit(X_val, y_val)
+    mapie_ts_reg.update(X_test, y_test, gamma=0.1, alpha=0.1)
+    assert mapie_ts_reg.method == method, \
+        f"Expected method {method}, but got {mapie_ts_reg.method}"
```
Comment (thibaultcordier) [inline mapie/tests/test_time_series_regression.py:542]: Here, all you have to do is check that the class has the right method. You don't have to handle any exceptions.
```suggestion
    assert mapie_ts_reg.method == method
```

File: mapie/tests/test_regression.py [hunk 4]
```diff
@@ -748,3 +748,20 @@ def test_predict_infinite_intervals() -> None:
...
+    X_train, X_val, y_train, y_val = train_test_split(
+        X, y, test_size=0.5, random_state=random_state
+    )
+    estimator = LinearRegression().fit(X_train, y_train)
+    mapie_reg = MapieRegressor(
+        cv=cv, method=method, estimator=estimator
+    )
+    mapie_reg.fit(X_val, y_val)
+    assert mapie_reg.method == "base", \
+        f"Expected method base, but got {mapie_reg.method}"
```
Comment (thibaultcordier) [inline mapie/tests/test_regression.py:767]: Here, all you have to do is check that the class has the right method. You don't have to handle any exceptions.
```suggestion
    assert mapie_reg.method == "base"
```

File: mapie/tests/test_regression.py [hunk 5]
```diff
@@ -748,3 +748,20 @@ def test_predict_infinite_intervals() -> None:
     _, y_pis = mapie_reg.predict(X, alpha=0., allow_infinite_bounds=True)
     np.testing.assert_allclose(y_pis[:, 0, 0], -np.inf)
     np.testing.assert_allclose(y_pis[:, 1, 0], np.inf)
+
+
+@pytest.mark.parametrize("method", ["minmax", "naive", "plus"])
```
Comment (thibaultcordier) [inline mapie/tests/test_regression.py:753]: You can add the `base` method to check that we preserve its value.
```suggestion
@pytest.mark.parametrize("method", ["minmax", "naive", "plus", "base"])
```

Comment (BaptisteCalot) [reply mapie/tests/test_regression.py:753]: Yes indeed, thank you Thibault for your remark !

File: HISTORY.rst [hunk 6]
```diff
@@ -4,7 +4,7 @@ History
 
 0.8.3 (2024-**-**)
 ------------------
-
+* Fix shift in power from one method to base method when use MapieRegressor fit : issue 447
```
Comment (thibaultcordier) [inline HISTORY.rst:7]:
```suggestion
* Fixed overloading of the value of the ‘method’ attribute when using MapieRegressor and MapieTimeSeriesRegressor
```

File: mapie/tests/test_regression.py [hunk 7]
```diff
@@ -748,3 +748,19 @@ def test_predict_infinite_intervals() -> None:
     _, y_pis = mapie_reg.predict(X, alpha=0., allow_infinite_bounds=True)
     np.testing.assert_allclose(y_pis[:, 0, 0], -np.inf)
     np.testing.assert_allclose(y_pis[:, 1, 0], np.inf)
+
+
+@pytest.mark.parametrize("method", ["minmax", "naive", "plus", "base"])
+@pytest.mark.parametrize("cv", ["split", "prefit"])
+def test_check_change_method_to_base(method: str, cv: str) -> None:
+    """Test of shift in power from one method to base method in fit"""
```
Comment (thibaultcordier) [inline mapie/tests/test_regression.py:756]:
```suggestion
    """Test of the overloading of method attribute to `base` method in fit"""
```

**Review by thibaultcordier — APPROVED** [review]
Thank for for your contribution @BaptisteCalot! I'll let @LacombeLouis check your code once more before merging it (+ suggested changes to the code form).

**Pull request merged**